torch/_inductor/triton_helpers.py

```diff
@@ -100,14 +100,50 @@ def maximum_with_index(a_value, a_index, b_value, b_index):
     return tl.where(mask, a_value, b_value), tl.where(mask, a_index, b_index)
 
 
-@triton.jit
-def min_with_index(value, index, dim):
-    return tl.reduce((value, index), dim, minimum_with_index)
+if TRITON_HAS_REDUCE:
 
+    @triton.jit
+    def min_with_index(value, index, dim):
+        return tl.reduce((value, index), dim, minimum_with_index)
 
-@triton.jit
-def max_with_index(value, index, dim):
-    return tl.reduce((value, index), dim, maximum_with_index)
+    @triton.jit
+    def max_with_index(value, index, dim):
+        return tl.reduce((value, index), dim, maximum_with_index)
+
+else:
+
+    @triton.jit
+    def _argreduce_index(reduction_result, value, index, dim):
+        reduction_result_keepdim = reduction_result[None, :]
+        if dim == 0:
+            pass
+        elif dim == 1:
+            reduction_result_keepdim = reduction_result[:, None]
+        else:
+            tl.device_assert(False)
+
+        equal = value == reduction_result_keepdim
+        if is_floating(value):
+            # Treat nan as equal
+            result_is_nan = reduction_result_keepdim != reduction_result_keepdim
+            equal |= (value != value) and result_is_nan
+
+        invalid_index = 2**62
+        indices = tl.where(equal, index, invalid_index)
+        index = tl.min(indices, dim)
```

Comment:
Here instead of masking to zero and summing the indices, I instead mask with a large value (`2**62`) and take the minimum. That should always return a valid index, while also having the behavior of taking the smallest index which matches eager.